File: app/src/main/java/de/squiray/dailytodo/domain/repository/TodoRepository.java
package de.squiray.dailytodo.domain.repository;

import java.util.List;

import de.squiray.dailytodo.domain.entity.Todo;

public interface TodoRepository {

    Todo save(Todo todo);

    Todo get(String id);

    List<Todo> getAll();

    void deleteTodo(Todo todo);
}

File: app/src/main/java/de/squiray/dailytodo/domain/repository/TodoRepository.kt
package de.squiray.dailytodo.domain.repository

import de.squiray.dailytodo.domain.entity.Todo

interface TodoRepository {

    val all: List<Todo>

    fun save(todo: Todo): Todo

    operator fun get(id: String): Todo

    fun deleteTodo(todo: Todo)
}
